src/server/cdb/data/process.c

```diff
@@ -0,0 +1,500 @@
+#include "process.h"
+
+#include "protocol/data/memcache_include.h"
+#include "storage/slab/slab.h"
+#include "storage/cdb/cdb.h"
+
+#include <cc_array.h>
+#include <cc_debug.h>
+#include <cc_print.h>
+
+#define CDB_PROCESS_MODULE_NAME "cdb::process"
+
+#define OVERSIZE_ERR_MSG    "oversized value, cannot be stored"
+#define DELTA_ERR_MSG       "value is not a number"
+#define OOM_ERR_MSG         "server is out of memory"
+#define CMD_ERR_MSG         "command not supported"
+#define OTHER_ERR_MSG       "unknown server error"
+
+
+typedef enum put_rstatus {
+    PUT_OK,
+    PUT_PARTIAL,
+    PUT_ERROR,
+} put_rstatus_t;
+
+static bool process_init = false;
+static process_metrics_st *process_metrics = NULL;
+static bool allow_flush = ALLOW_FLUSH;
+
+static struct CDBHandle *cdb_handle = NULL;
+
+void
+process_setup(process_options_st *options, process_metrics_st *metrics, struct CDBHandle *handle)
+{
+    log_info("set up the %s module", CDB_PROCESS_MODULE_NAME);
+
+    if (process_init) {
+        log_warn("%s has already been setup, overwrite",
+                 CDB_PROCESS_MODULE_NAME);
+    }
+
+    process_metrics = metrics;
+
+    if (options != NULL) {
+        allow_flush = option_bool(&options->allow_flush);
+    }
+
+    if (handle == NULL) {
+        log_crit("handle argument was null! wth?!");
+        // TODO(simms): how do i die here?
+    }
+    cdb_handle = handle;
+
+    process_init = true;
+}
+
+void
+process_teardown(void)
+{
+    log_info("tear down the %s module", CDB_PROCESS_MODULE_NAME);
+    if (!process_init) {
+        log_warn("%s has never been setup", CDB_PROCESS_MODULE_NAME);
+    }
+
+    if (cdb_handle != NULL) {
+        struct CDBHandle *p = cdb_handle;
+        cdb_handle = NULL;
+        cdb_handle_destroy(p);
+    }
+
+    allow_flush = false;
+    process_metrics = NULL;
+    process_init = false;
+}
+
+
+
+static bool
+_get_key(struct response *rsp, struct bstring *key)
+{
+    /* this is a slight abuse of the bstring API. we're setting
+     * the data pointer to point to the vbuf that was allocated on the struct
+     * and we're setting the len of the buffer to the allocation size. This is
+     * so that we can create a rust slice from this information. */
+    rsp->vstr.data = rsp->vbuf;
+    rsp->vstr.len = RSP_VAL_BUF_SIZE;
```

Comment:
This is where the newly added `rsp->vbuf` gets hooked up for use on the rust side.